hangman.py

```diff
@@ -0,0 +1,348 @@
+                if letters_guessed in vowels_list:
+                    tries -= 2
+                else:
+                    tries -= 1
+                check = get_guessed_word(secret_word, letters_guessed_list)
+                print("Oops! That letter is not in my word:", check)
+            letters.append(letters_guessed)
+        elif len(letters_guessed) < 1:
+            if warnings > 0:
+                warnings -= 1
+            else:
+                warnings = "no"
+                text = " so you lose one guess"
+                tries -= 1
+            print("Oops! Don't leave a blank line. You have", warnings, "warnings left:", check)
+        else:
+            if warnings > 0:
+                warnings -= 1
+            else:
+                warnings = "no"
+                text = " so you lose one guess"
+                tries -= 1
+            print("Oops! Don't enter more than one symbol. You have", warnings, "warnings left:", check)
+        if tries == 0:
+            print("-" * 13)
+            print('Sorry, you ran out of guesses. The word was "' + secret_word + '".')
+            break
+        if secret_word == check:
+            score = tries * len(set(secret_word))
+            print("-" * 13)
+            print("Congratulations, you won! Your total score for this game is:", score)
+            break
+    return
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    my_word = "".join(my_word.split())
+    if len(my_word) == len(other_word):
+        i = 0
+        while i < len(my_word):
+            if my_word[i] == "_":
+                i += 1
+            else:
+                if my_word[i] != other_word[i]:
+                    return False
+                else:
+                    i += 1
+        return True
+    else:
+        return False
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+
+    '''
+    matches = [word for word in wordlist if match_with_gaps(my_word, word) == True]
+    if not matches:
+        print("No matches found.\n")
+    else:
+        print("Possible word matches are: " + ", ".join(matches), end="." + "\n")
+    return
+
+
+def hangman_with_hints(secret_word):
```

Comment:
Effectively this code is mostly the same as code in function `hangman`. Try to avoid code duplication - combine these two functions